discovery-provider/src/queries/notifications.py

```diff
@@ -795,7 +795,7 @@ def notifications():
             )
             # Previous private entry indicates transition to public, triggering a notification
             prev_entry = prev_entry_query.first()
-            if prev_entry.is_private == True:
+            if prev_entry and prev_entry.is_private == True:
```

Comment:
does the prev_entry_query still work here? shouldn't we change fetching prev entry to fetch from revert blocks table?